docs/advanced-search.md

```diff
@@ -0,0 +1,339 @@
+# Advanced Search
+
+```eval_rst
+.. contents:: :local:
+```
+
+Kitsune supports an advanced search syntax in all its search boxes.
+
+A number of search tokens,
+described below,
+take a `{field_name}` as an argument.
+The valid field names for each of our document types are explained in the [Document Fields](#document-fields) section.
+
+## Quoted Phrase
+
+`"x y z"`
+
+Matches the phrase,
+disabling synonym matching.
+
+e.g. `add-ons NOT "themes"`
+will match add-ons and extensions, but not themes.
+
+## Field Operator
+
+`field:{field_name}:{query}`
+
+Specifies that `{query}` should be run against `{field_name}`.
+If enclosed in brackets,
+`{query}` can have nested operators.
+
+e.g. `field:keywords.en-US:(firefox NOT android)`
+will find KB documents with `firefox` and without `android` in their keywords field.
+
+### Aliases
+
+Each document type has a number of aliases,
+shown below,
+which allows easy searching across document types.
+
+e.g. using `field:title:firefox` in instant search would find both articles and questions with firefox in the title.
+
+#### Knowledge Base Articles
+
+```eval_rst
+.. list-table::
+    :header-rows: 1
+
+    * - Alias
+      - Maps To
+    * - ``title``
+      - ``title.{locale}``
+    * - ``content``
+      - ``content.{locale}``
+```
+
+#### Questions
+
+```eval_rst
+.. list-table::
+    :header-rows: 1
+
+    * - Alias
+      - Maps To
+    * - ``title``
+      - ``question_title.{current_locale}``
+    * - ``content``
+      - ``question_content.{current_locale}`` or ``answer_content.{current_locale}``
+    * - ``question``
+      - ``question_content.{current_locale}``
+    * - ``answer``
+      - ``answer_content.{current_locale}``
+```
+
+#### Forum Posts
+
+```eval_rst
+.. list-table::
+    :header-rows: 1
+
+    * - Field
+      - Maps To
+    * - ``title``
+      - ``thread_title``
+```
+
+## Boolean Operators
+
+### NOT
+
+`NOT x`
+
+Specifies that the token must not exist in the document.
+
+### AND
+
+`x AND y`
+
+Specifies that both tokens must exist in the document.
+
+```eval_rst
+.. note::
+    A quirk of our current implementation is that ``x AND y`` can return more results than ``x y``.
```

Comment:
Maybe it's worth mentioning that `x y` is also `AND` search but the difference is the requirement of existence in the same field for the latter. I am proposing this b/c usually users expect `OR` operator as the default. It wouldn't matter without the advanced syntax but now it could help avoiding confusion.